src/lib/addons/datadog.ts

```diff
@@ -113,16 +113,16 @@ export default class DatadogAddon extends Addon {
             state = 'failed';
             const failedMessage = `Datadog Events API request failed with status code: ${res.status}.`;
             stateDetails.push(failedMessage);
-            if (this.flagResolver.isEnabled('addonUsageMetrics')) {
-                this.eventBus.emit(ADDON_EVENTS_HANDLED, {
-                    result: state,
-                    destination: 'datadog',
-                });
-            }
-
             this.logger.warn(failedMessage);
         }
 
+        if (this.flagResolver.isEnabled('addonUsageMetrics')) {
+            this.eventBus.emit(ADDON_EVENTS_HANDLED, {
+                result: state,
+                destination: 'datadog',
+            });
+        }
```

Comment:
This will work, but I'm thinking, maybe it's best to go with Ivar's approach... if in the next line we are registering an event, maybe we can rely on a listener that listens all events but only counts the ones that have an `integrationId` property set...